grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java

```diff
@@ -191,8 +191,16 @@
 
         this.res = requireNonNull(res, "res");
         this.compressorRegistry = requireNonNull(compressorRegistry, "compressorRegistry");
-        clientAcceptEncoding =
-                Strings.emptyToNull(clientHeaders.get(GrpcHeaderNames.GRPC_ACCEPT_ENCODING));
+        clientAcceptEncoding = clientHeaders.get(GrpcHeaderNames.GRPC_ACCEPT_ENCODING, "");
+        if (autoCompression && !clientAcceptEncoding.isEmpty()) {
+            final Iterable<String> acceptedEncodings = ACCEPT_ENCODING_SPLITTER.split(clientAcceptEncoding);
```

Comment:
The `clientAcceptEncoding` split twice in certain circumstances: https://github.com/line/armeria/blob/master/grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java#L260
Also, It's a bit confusing to me the relationship between `messageCompression` and `autoCompression`.
So, how about splitting and setting the compressor in `doSendHeaders` as we do for now?
```java
if (messageCompression && clientAcceptEncoding != null) {
    if (compressor != null) {
        ...
    } else if (autoCompress) {
        ...
    } else {
        compressor = Codec.Identity.NONE;
    }
} else {
    compressor = Codec.Identity.NONE;
}
```

Reply:
It seems good!